components/brave_wallet/browser/swap_response_parser.cc

```diff
@@ -472,7 +471,22 @@ mojom::LiFiStepEstimatePtr ParseEstimate(
     result->gas_costs.push_back(std::move(gas_cost));
   }
 
-  result->execution_duration = value.execution_duration;
+  if (!value.fee_costs) {
+    return result;
+  }
+
+  std::vector<mojom::LiFiFeeCostPtr> fee_costs = {};
+  for (const auto& fee_cost_value : *value.fee_costs) {
+    auto fee_cost = mojom::LiFiFeeCost::New();
+    fee_cost->name = fee_cost_value.name;
+    fee_cost->description = fee_cost_value.description;
+    fee_cost->percentage = fee_cost_value.percentage;
+    fee_cost->token = ParseToken(fee_cost_value.token);
+    fee_cost->amount = fee_cost_value.amount;
+    fee_cost->included = fee_cost_value.included;
+    fee_costs.push_back(std::move(fee_cost));
+  }
+  result->fee_costs = std::move(fee_costs);
```

Comment:
Let's not do early return here. Someone might add some parsing after this block which would be incorrectly lost when there is no fee_costs
So this way is preferred
```
  if (value.fee_costs) {
    for (const auto& fee_cost_value : *value.fee_costs) {
      ...
      result->fee_costs.push_back(std::move(fee_cost));
    }
  }
```